From c405d47d24c5e17cfec21b73133debd222f03960 Mon Sep 17 00:00:00 2001
From: Akira Ajisaka <aajisaka@apache.org>
Date: Fri, 15 Jul 2016 11:42:44 -0700
Subject: [PATCH 1943/2748] HDFS-10628. Log HDFS Balancer exit message to its
 own log. Contributed by Jiayi Zhou.

Conflicts:
	hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/balancer/Balancer.java

(cherry picked from commit c48e9d608c1daf7826f2e738c0a67d8eab79d8de)

Change-Id: I068ac33eb53ddd7a62008b36dffacd7cd267dc20
---
 .../hadoop/hdfs/server/balancer/Balancer.java      |    2 ++
 1 file changed, 2 insertions(+)

diff --git a/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/balancer/Balancer.java b/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/balancer/Balancer.java
index 4a682e1..73d7f38 100644
--- a/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/balancer/Balancer.java
+++ b/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/balancer/Balancer.java
@@ -581,6 +581,8 @@ Result runOneIteration() {
       if (!runDuringUpgrade && nnc.isUpgrading()) {
         System.err.println("Balancer exiting as upgrade is not finalized, "
             + "please finalize the HDFS upgrade before running the balancer.");
+        LOG.error("Balancer exiting as upgrade is not finalized, "
+            + "please finalize the HDFS upgrade before running the balancer.");
         return newResult(ExitStatus.UNFINALIZED_UPGRADE, bytesLeftToMove, -1);
       }
 
-- 
1.7.9.5

